src/k8s-engine/wsl.ts

```diff
@@ -421,7 +387,10 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
       return;
     }
     if (semver.gt(existingVersion, desiredVersion)) {
-      await this.k3sHelper.deleteKubeState((...args) => this.execCommand(...args));
+      await this.progressTracker.action(
+        'Deleting incompatible Kubernetes state',
+        100,
+        this.k3sHelper.deleteKubeState((...args) => this.execCommand(...args)));
```

Comment:
There is a missing `await` when deleting the kube state. See the removed line.

This causes downgrading to not work. In my wsl logs I found...

```
rm: can't remove '/var/lib/kubelet/pods/1938ac0a-2045-4c3e-aba2-9711982ead31/volumes/kubernetes.io~projected/kube-api-access-wc9b9': Resource busy
rm: can't remove '/var/lib/kubelet/pods/21e88a7a-e320-4fc2-a0b2-f847be1b7386/volumes/kubernetes.io~projected/kube-api-access-6pxcm': Resource busy
rm: can't remove '/var/lib/kubelet/pods/43b82490-bdd7-473d-a45e-4895e37eccba/volumes/kubernetes.io~secret/certs-server': Resource busy
rm: can't remove '/var/lib/kubelet/pods/43b82490-bdd7-473d-a45e-4895e37eccba/volumes/kubernetes.io~secret/certs-ca': Resource busy
rm: can't remove '/var/lib/kubelet/pods/43b82490-bdd7-473d-a45e-4895e37eccba/volumes/kubernetes.io~projected/kube-api-access-vj8mv': Resource busy
rm: can't remove '/var/lib/kubelet/pods/46434c39-6ccf-47bf-9ee5-97b08ce77f4a/volumes/kubernetes.io~projected/kube-api-access-5nhtb': Resource busy
rm: can't remove '/var/lib/kubelet/pods/599c74ff-6c0f-42f2-909e-e08471a86099/volumes/kubernetes.io~projected/kube-api-access-tnclj': Resource busy
rm: can't remove '/var/lib/kubelet/pods/2d8ae630-5c6e-4d5e-a52b-b95ef9cdabe9/volumes/kubernetes.io~projected/kube-api-access-gl6bl': Resource busy
WSL failed to execute rm -rf /var/lib/kubelet
```

Reply:
The last argument to `action` is either a `Promise` (which will be waited on before `action` returns), or a nullary function returning a `Promise` (in which case the function is invoked, and the result is waited on before `action` returns).

The problem here appears to be that we still have a pile of bind mounts (from Kubernetes) lying around, and that's leading to issues.  Attempting to unmount them manually, but it doesn't seem to like me (`Invalid argument`)…

Reply:
Since this appears to be an issue that already exists on `origin/main`, I'll spawn this as a new bug (but I'm still looking at it).

Reply:
I was able to duplicate the `busy resource` error on both this branch and `main`, so I agree it's a separate problem.

Otherwise this looks good. I was working on the sister issue #660 and this looks like a better solution.

Reply:
Ah yes, I see it on main as well.